app/models/concerns/profile_policy_scoring.rb

```diff
@@ -9,4 +9,34 @@ def score(host: nil)
 
     ((scores = results.pluck(:score)).sum / scores.size).to_f
   end
+
+  def update_counters!
+    # rubocop:disable Rails/SkipsModelValidations
+    update_columns(
+      total_host_count: hosts.count,
+      test_result_host_count: latest_supported_policy_test_result_hosts.count,
+      compliant_host_count: latest_supported_policy_test_result_hosts
+        .count { |host| compliant?(host) },
+      unsupported_host_count: latest_unsupported_policy_test_result_hosts.count
+    )
+    # rubocop:enable Rails/SkipsModelValidations
+  end
+
+  private
+
+  def latest_supported_policy_test_result_hosts
+    ::Host.where(id: latest_policy_test_results.supported.select(:host_id))
```

Comment:
Technically, we shouldn't need to query hosts for this count. Since we're within a single policy, it should be the same as the number of latest supported test results. If you want to be sure, you can use `count(:host_id)`.

Reply:
We no longer need this as I went with the `has_many :through`

Reply:
Yeah, gonna simplify a bit

Reply:
Nope, it won't work :disappointed: we need the host because of the `compliant?` call